In [26]:
from openai import OpenAI
import os
import base64
import requests

from dotenv import load_dotenv
# Load API key
_ = load_dotenv()

In [27]:
def encode_image(image_path):
  with open(image_path, "rb") as image_file:
    return base64.b64encode(image_file.read()).decode('utf-8')

In [28]:
client = OpenAI()

In [29]:
prompt = '''

Brainstorm a list of product names for a plantain chips that promotes low fat low carb. Target Audience: 1) who don't know plaintain chips, 2) who are familiar with plaintain chips]

Return the results as a comma-separated list, in this format:
Product Description: [come up with one most catchy description that fits to all target audience; you may put emphasis on ingredients, calories, price, etc]
Product names: [list of 3 product names for all target audience]
Product slogan: [come up with one most catchy description that fits to all target audience]
Product social media post: [instagram version for each 2 target audience]

## Example1:
Product Description: Made from hand-selected plantains, these guilt-free chips pack only 90 calories per serving with zero artificial ingredients. Naturally gluten-free, low-carb, and crafted to perfection - bringing you tropical indulgence without compromising your health goals.

Product names: ThinTain, LightCrunch, TropicLite

Product slogan: "Crunch your way to happiness, one light bite at a time"

Social media posts:

[Instagram for newcomers]:
"🌟 Meet your new snack obsession!
At just 90 calories per serving, ThinTain brings you the perfect blend of crispy and light. Made from real plantains (nature's better banana! 🍌), these chips are your ticket to guilt-free snacking.

✨ Only 90 calories
🌱 100% natural ingredients
💪 Low carb & gluten-free
🌟 Perfect for any diet

Try something new today - your taste buds will thank you!
#HealthySnacking #NewSnackAlert #CleanEating"

[Instagram for familiar users]:
"🌿 Your favorite plantain chips just got a healthy upgrade!

Remember that irresistible plantain chip crunch you love? We've kept all the authentic flavor while cutting the calories to just 90 per serving! No compromises, just smart snacking.

✅ Same great taste
📉 Lower carbs
⚡️ Less fat
❤️ More satisfaction

Available now at [store link in bio]
#PlantainLovers #HealthySwaps #SnackBetter"

## Example2:

Product Description: Made from hand-selected plantains, these guilt-free chips pack only 90 calories per serving with zero artificial ingredients. Naturally gluten-free, 40% less fat than regular chips, and packed with potassium - your perfect everyday snack companion.

Product names: ThinCrisp, PlantainLite, TropicalThins

Product slogan: "Crunch your way to wellness"

Social media posts:

[Instagram for newcomers]:
"🌟 Meet your new snack obsession!
Only 90 calories per serving 🔥
✨ Naturally gluten-free
✨ 40% less fat than regular chips
✨ Rich in potassium
Swipe up to discover why everyone's talking about ThinCrisp - the light & crispy tropical snack that's changing the game! 🌱
.
.
#HealthySnacking #NewSnackAlert #CleanEating #GlutenFree #SnackTime"

[Instagram for familiar users]:
"👋 Hey plantain lovers!
Your favorite snack just got a healthy upgrade:
✅ Same authentic tropical taste
✅ 40% less fat
✅ Only 90 calories per serving
✅ No artificial ingredients

Time to stock up on the new ThinCrisp - because why choose between taste and health when you can have both? 💪
.
.
#PlantainChips #HealthySnacking #CleanEating #SnackBetter #PlantainLove"

'''

In [30]:
 messages=[
            {
                "role": "system",
                "content": "You are a helpful assistant."
            },
            {
                "role": "user",
                "content": prompt
            }
        ]

In [31]:

completion = client.chat.completions.create(
  model="gpt-4o",
  messages=messages
)

response = completion.choices[0].message.content

In [32]:
num_of_repeats = 3

responses = []

for qcnt, query in enumerate(range(num_of_repeats),1):
    completion = client.chat.completions.create(
      model="gpt-4o",
      messages=messages
    )

    response = completion.choices[0].message.content

    responses.append(response)

    print(f'\n--- {qcnt} of {num_of_repeats} ---\n{response}')


--- 1 of 3 ---
Product Description: Dive into the world of crispy delight with our new plantain chips! Each serving comes in at just 80 calories, boasting all-natural ingredients with 35% less fat and carbs compared to regular chips. Enjoy the authentic taste with none of the guilt—perfectly crafted for every snack enthusiast.

Product names: CrispTain, LeanPlantains, TropiCrunch

Product slogan: "Satisfy your crunch cravings without compromise"

Social media posts:

[Instagram for newcomers]:
"🌟 Discover the snack that’s taking the world by storm—at only 80 calories!
Our CrispTain chips are crafted from the finest plantains (nature's savory bite 😉) for a low-fat, low-carb crunch that's irresistibly delicious.

✨ Just 80 calories
🌱 Pure, natural ingredients
🥥 Tropical taste in every crunch
Perfect snack for any time of the day!

Explore a new world of flavor—you'll never look at chips the same way again!
#HealthySnacking #PlantainDiscovery #SnackSmart #NewTaste"

[Instagram for famili

<img src="logo1.png"/>

--- 1 of 3 ---
Product Description: Dive into the world of crispy delight with our new plantain chips! Each serving comes in at just 80 calories, boasting all-natural ingredients with 35% less fat and carbs compared to regular chips. Enjoy the authentic taste with none of the guilt—perfectly crafted for every snack enthusiast.

Product names: CrispTain, LeanPlantains, TropiCrunch

Product slogan: "Satisfy your crunch cravings without compromise"

Social media posts:

[Instagram for newcomers]:
"🌟 Discover the snack that’s taking the world by storm—at only 80 calories!
Our CrispTain chips are crafted from the finest plantains (nature's savory bite 😉) for a low-fat, low-carb crunch that's irresistibly delicious.

✨ Just 80 calories
🌱 Pure, natural ingredients
🥥 Tropical taste in every crunch
Perfect snack for any time of the day!

Explore a new world of flavor—you'll never look at chips the same way again!
#HealthySnacking #PlantainDiscovery #SnackSmart #NewTaste"

[Instagram for familiar users]:
"🌿 Plantain enthusiasts, meet your new favorite snack!
Crispy goodness with 35% less fat—our CrispTain chips deliver the authentic taste you love, now even more guilt-free.

✅ Same iconic flavor
🔥 Only 80 calories per serving
🌿 Natural, simple ingredients
Upgrade your snack game today!

Why settle for ordinary when you can indulge smartly? 🍃
#PlantainPerfection #HealthyHabits #CrispAndLight #SnackSatisfaction"



<img src="logo2.png"/>

--- 2 of 3 ---
Product Description: Introducing our plantain chips, made from naturally low-carb, premium plantains and containing only 80 calories per serving. With zero added sugars and 35% less fat than traditional chips, they're the perfect mix of wholesome nutrition and delicious tropical flavor.

Product names: PlantainLite, SlimCrunch, TropiBites

Product slogan: "Savor the crunch, love the health."

Social media posts:

[Instagram for newcomers]:
"🌟 Discover your new favorite snack! 
Our PlantainLite chips are the answer to your snacking prayers: 80 calories of crisp goodness per serving, packed with all-natural ingredients. 

🌿 100% natural plantains
🍌 No added sugars
💪 35% less fat than regular chips
🌟 Perfectly light & satisfying

Dare to snack differently today!
#NewSnackObsession #HealthyChoices #TrySomethingNew"

[Instagram for familiar users]:
"🍃 Hey plantain chip fans!
The snack you love, now even better. Get ready for PlantainLite - your go-to for a healthy crunch with 35% less fat and only 80 calories per serving.

✔️ All the exotic taste you adore
✔️ Smarter snacking, fewer calories
✔️ Naturally delicious, naturally gluten-free

Elevate your snack game now!
#PlantainPassion #SmartSnacking #SnackRevolution"